frontend/src/components/nodes/NodeNav.tsx

```diff
@@ -3,138 +3,107 @@ import Link from "next/link";
-}
-
-export default class extends React.Component<Props, State> {
-  constructor(props: any) {
-    super(props);
-    this.timer = null;
-  }
-
-  timer: ReturnType<typeof setTimeout> | null;
-  state: State = {};
-
-  componentDidMount() {
-    this.timer = setTimeout(this.regularFetchInfo, 0);
-  }
-
-  componentWillUnmount() {
-    const timer = this.timer;
-    this.timer = null;
-    if (timer !== null) {
-      clearTimeout(timer);
-    }
-  }
-
-  regularFetchInfo = async () => {
-    await this.fetchNodes();
-    if (this.timer !== null) {
-      this.timer = setTimeout(this.regularFetchInfo, 10000);
-    }
-  };
-
-  fetchNodes = async () => {
-    new NodesApi()
-      .getOnlineNodesStats()
-      .then((nodeStats) => this.setState({ nodeStats }));
-  };
 
+export default class extends React.Component<Props> {
   render() {
-    const { nodeStats } = this.state;
     const { role } = this.props;
     return (
-      <>
-        <Row>
-          <Link href="/nodes/validators">
-            <a
-              className={`node-link ${
-                role === "validators" ? `node-selected` : ""
-              }`}
-              id="validator-node"
-            >
-              <Col className="node-selector align-self-center">
-                {nodeStats
-                  ? `${nodeStats.validatorsCount}  Validating`
-                  : `- Validating`}
-              </Col>
-            </a>
-          </Link>
-          <Link href="/nodes/online-nodes">
-            <a
-              className={`node-link ${
-                role === "online-nodes" ? `node-selected` : ""
-              }`}
-              id="online-node"
-            >
-              <Col className="node-selector align-self-center">
-                {nodeStats
-                  ? `${nodeStats.onlineNodesCount} Online-nodes`
-                  : `- Online-nodes`}
-              </Col>
-            </a>
-          </Link>
-          <Link href="/nodes/proposals">
-            <a
-              className={`node-link ${
-                role === "proposals" ? `node-selected` : ""
-              }`}
-              id="proposal-node"
-            >
-              <Col className="node-selector align-self-center">
-                {nodeStats
-                  ? `${nodeStats.proposalsCount} Proposal-nodes`
-                  : `- Proposal-nodes`}
-              </Col>
-            </a>
-          </Link>
-          <Link href="/nodes/map">
-            <a className="node-link">
-              <Col className="node-selector align-self-center">Nodes Map</Col>
-            </a>
-          </Link>
-        </Row>
-        <style jsx global>{`
-          .node-selector {
-            font-size: 12px;
-            font-weight: 500;
-            letter-spacing: 1.38px;
-            color: #24272a;
-            text-transform: uppercase;
-            padding: 10px;
-            text-decoration: none;
-          }
+      <NodeStatsConsumer>
+        {(context) => (
```

Comment:
In some cases, you may consider unpacking the context immediately. I don't have a strong preference, just wanted to mention.

```suggestion
        {({ nodeInfo }) => (
```